poly-synth/src/poly-synth-voice.js

```diff
@@ -58,12 +58,35 @@ class PolySynthVoice {
   start() {
     // Ramp to full amplitude in attack (s) and to sustain in decay (s).
     const t = this.context_.currentTime;
-    const timeToFullAmplitude = t + this.parameters_.attack;
-    const timeToSustain = timeToFullAmplitude + this.parameters_.decay;
+    const timeToFullAmplitude = t + this.parameters_.gainAttack;
+
+    // |timeToGainSustain| must be later than |timeToFullAmplitude|.
+    const timeToGainSustain =
+        timeToFullAmplitude + this.parameters_.gainDecay + 0.001;
+
     this.output.gain.setValueAtTime(0, t);
     this.output.gain.linearRampToValueAtTime(1, timeToFullAmplitude);
     this.output.gain.linearRampToValueAtTime(
-        this.parameters_.sustain, timeToSustain);
+        this.parameters_.gainSustain, timeToGainSustain);
+
+    // The detune of the filter reaches its peak amount specified by
+    // |filterDetuneAmount| (where 1 corresponds to 2400 cents detuning) in
+    // |filterAttack| seconds. It then decays to a fraction of that amount as
+    // specified by |filterSustain| in |filterDecay| seconds.
+    const standardFilterDetuneInCents = 2400;
+    const amountOfFilterDetuneInCents =
+        standardFilterDetuneInCents * this.parameters_.filterDetuneAmount;
```

Comment:
The convention according to @hoch is to let `filterDetuneAmount` be a multiplier of a standard detune amount (2400). So when `filterDetuneAmount` is 0, there is no detuning and when it is its maximum value 5, the detuning is 1200 (5 *2400).